From c9d8e75c6469088e9daa4f8741a459594e7d63e8 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Sat, 21 Apr 2018 01:49:41 -0700
Subject: [PATCH] dhcpv6: Fix strncpy bounds and initialize struct to 0

Fixes
dhcpv6.c:141:2: error: 'strncpy' specified bound 16 equals destination size [-Werror=stringop-truncation]                                                                                                        strncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));

Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/dhcpv6.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

Index: git/src/dhcpv6.c
===================================================================
--- git.orig/src/dhcpv6.c
+++ git/src/dhcpv6.c
@@ -135,7 +135,7 @@ int init_dhcpv6(const char *ifname, unsi
 	// Detect interface
 	struct ifreq ifr;
 	memset(&ifr, 0, sizeof(ifr));
-	strncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name));
+	strncpy(ifr.ifr_name, ifname, sizeof(ifr.ifr_name)-1);
 	if (ioctl(sock, SIOCGIFINDEX, &ifr) < 0)
 		goto failure;
 
